### DISCUSSION TITLE
Code syntax highlight library

### DISCUSSION THREAD

**Discussion by ivanjaros** [markedjs/marked#3401] [Q&A] [answered]
What code syntax highlight **extension** would you recommend? I tried the https://www.npmjs.com/package/marked-highlight but it's not working(maybe it needs to be injected directly into marked and not as an extension, which does not work for me). Is there anything else available?

**Comment by UziTech** [comment]
Why is it not working? That is the one used in [marked docs](https://github.com/markedjs/marked/blob/master/docs/build.js#L5).

**Comment by ivanjaros** [comment]
Nevermind, I tried marked-shiki but it did not work either. I guess they both require a special way to be initialized and not used as mere extensions, which does not work for my setup. I will probably move to marked-it instead.

**Reply by UziTech** [reply]
We use marked-highlight as an extension. I don't know what special way you are talking about.

**Reply by ivanjaros** [reply]
Well, it's not an extension. It forces itself to be a markdown constructor that overrides some methods. Extension can be simply added into the "extensions" list when initiating markdown instance.

**Reply by ivanjaros** [reply]
For example my markdown is being initiated like so:
```
new Marked({
    async: true,
    breaks: true,
    gfm: true,
    extensions: [foo, bar],
    walkTokens: async (token) => { ... }
```

Highlight forces itself to be injected as `new Marked(markedHighlight({...}))`.

**Reply by UziTech** [reply, marked as answer]
No you can use `marked.use` to use extensions after instantiation.

```js
const marked = new Marked({
    async: true,
    breaks: true,
    gfm: true,
    extensions: [foo, bar],
    walkTokens: async (token) => { ... }
});

marked.use(markedHighlight({...}));
```

You can also add multiple extensions to new Marked

```js
const marked = new Marked({
    async: true,
    breaks: true,
    gfm: true,
    extensions: [foo, bar],
    walkTokens: async (token) => { ... }
},
markedHighlight({...}));
```
